src/sonymn_int.hpp

```diff
@@ -71,6 +75,11 @@ namespace Exiv2 {
         static const TagInfo tagInfoFp_[];
         static const TagInfo tagInfoSonyMisc1_[];
         static const TagInfo tagInfo2010e_[];
+
+        // TODO: When moving to C++20, this can be replace with
+        //       std::basic_string<CharT,Traits,Allocator>::starts_with() .
+        //       Suggested in https://github.com/Exiv2/exiv2/pull/1777 .
+        static bool startsWith(const std::string& s, const std::string& start);
```

Comment:
I wouldn't put this in the header file, since this function is only intended to be used in one source file (`sonymn_int.cpp`). It looks like you need to move the definition of `startWith` to the top of the cpp file. It is currently on line 958, which is after the use on line 867.

Reply:
I agree, I tried adding the function in the way that you suggest. The problem was that the `startsWith()` function in `makernote_int.cpp` conflicted when linking - both definitions were in the same namespace.
I attempted adding the definition to `makernote_int.hpp`, but this didn't imediately work and I was trying to avoid adding includes to `sonymn_int.cpp`.

I am open to suggestions.

Reply:
Got it, sorry about that. Ok, it looks to me like the best plan is to add it to `src/utils.hpp` and `src/utils.cpp`. Please could you remove the copy from `epsimage.cpp` too and get all three files to include "utils.hpp"? Thanks!

Reply:
No problem, thanks for the help.
I have made the changes that you suggested.